### ISSUE TITLE
Use generated message as a default one in sourceLocale catalog

### ISSUE THREAD

**Issue by sladek-jan** [lingui/js-lingui#212]
Hi,

In an attempt to add metadata ("context" for linguists) to translations, I used this [new feature](https://lingui.github.io/js-lingui/releases/migration-2.html#new-features) as follows:
`` i18n('Word // some context here')`Word` ``

The docs say `Generated message is used as a default one`. If I understand it correctly, in this case a key `Word // some context here` should be created in the catalog, with a default English translation `Word` (as long as sourceLang="en").

This is not what happens. The entry created in the catalog for `en` is:
``` 
"Word // some context here": {
    "translation": "Word // some context here",
    "origin": [
     ...
    ]
  },
```

This is logical, because of [this](https://github.com/lingui/js-lingui/pull/147/commits/d6be6be41c06e3be565a091bb818a000a4e08439) commit which copies translation IDs over as default translations into source locale. **But isn't this in conflict with what the docs say?**

I am currently using version 2.0.1 until [another](https://github.com/lingui/js-lingui/issues/171) issue is resolved.
Thank you for the great lib btw.

**Comment by snegostup** [comment]
Yeah, using messages as IDs can be tough :) I believe this can be fixed changing [this](https://github.com/lingui/js-lingui/blob/34b866940e9de832b3b52928361d456818470746/packages/cli/src/api/formats/lingui.js#L64) line to
```javascript
translation: config.sourceLocale === locale ? message.defaults : "",
```
as I proposed [here](https://github.com/lingui/js-lingui/issues/199)

**Comment by tricoder42** [comment]
@sladek-jan I've just created a PR and it should be merged soon. 

Just curious: if there were a better way to add metadata to translations, would you still use custom IDs?

For example adding metadata to comments:

```js
// i18n: some context here
i18n.t`Word`
```

Generated message catalog might look like this:

```
"Word": {
    "translation": "",
    "context": "some context here",
    "origin": [
     ...
    ]
  },
```

Just an idea I had in the past.

**Comment by sladek-jan** [comment]
@tricoder42 Ok, thank you! 

I guess it depends on what such a way would look like exactly.
Comments sound reasonable, and it would certainly look cleaner than placing the context after two slashes. 
What I like most about i18n is the conciseness of use though, so if there was a way to place such context directly into code (similarly as it works with custom IDs), I would find it even better. Also, it might spare some code crawling, I imagine. 

But I find comments just fine too.

**Comment by tricoder42** [comment]
Adding context directly to code might be even better. We just have to  remember to remove it from production bundle as context is just for documentation purposes (for translators).

I'll think about it, what solution is easier.

**Comment by tricoder42** [comment]
Fixed and released in [v2.2.0](https://github.com/lingui/js-lingui/blob/stable-2.x/CHANGELOG.md#220-2018-07-04). Thanks to all involved!
